library/edi.inc

```diff
@@ -768,6 +649,459 @@ function getUsernames()
 function arrFormated(&$item, $key)
 {
     $item = strstr($item, '_');
-    $item = substr($item, 1, strlen($item)-1);
-    $item = "'".$item;
+    $item = substr($item, 1, strlen($item) - 1);
+    $item = "'" . $item;
+}
+
+function requestEligibility($partner = '', $x12_270 = '')
+{
+    global $X12info;
+    if (((int)$X12info['id'] !== (int)$partner) && (int)$partner > 0) {
+        $X12info = getX12Partner($partner);
+    }
+
+    $payloadId = "3b8c13f5-11e2-43bf-bc47-737cca04f3fe"; // a default fallback
+    if (function_exists('openssl_random_pseudo_bytes') === true) {
+        $data = openssl_random_pseudo_bytes(16);
+        $data[6] = chr(ord($data[6]) & 0x0f | 0x40);
+        $data[8] = chr(ord($data[8]) & 0x3f | 0x80);
+        $payloadId = vsprintf('%s%s-%s-%s-%s-%s%s%s', str_split(bin2hex($data), 4));
+    }
+
+    $boundary = createUniqueToken(8);
```

Comment:
Just to clarify this will create 8 random bytes that is then converted to base64, which should become 11 characters (if my math is correct :) )

Reply:
btw, there will soon be a function in codebase (part of MFA PR) that will randomly create a string of desired length with desired characters.

Reply:
it's fine. this is used to create boundary for parsing.